vllm/engine/llm_engine.py

```diff
@@ -575,26 +577,51 @@ def _get_stats(self,
         # Iteration stats if we have scheduler output.
         num_prompt_tokens = 0
         num_generation_tokens = 0
+        num_prompt_tokens_lst = []
+        num_generation_tokens_lst = []
+        request_n = []
+        request_best_of = []
         time_to_first_tokens = []
         time_per_output_tokens = []
         time_e2e_requests = []
+        finished_reason_counter = CollectionsCounter()
         if scheduler_outputs is not None:
             prompt_run = scheduler_outputs.num_prefill_groups > 0
 
-            # Number of Tokens.
+            # Number of Tokens
             if prompt_run:
-                num_prompt_tokens = sum(
+                num_prompt_tokens_lst = [
                     len(scheduled_seq_group.seq_group.prompt_token_ids)
```

Comment:
Actually now that I think about it, we have a bug on current main for this same reason

We are actually processing `chunk_len` tokens rather then `len(prompt_token_ids`) in this iteration